api/topic.go

```diff
@@ -19,18 +19,55 @@ func (api *API) GetTopicHandler(w http.ResponseWriter, req *http.Request) {
 		handlers.CollectionID.Header(): hColID,
 		"request_id":                   ctx.Value(dprequest.RequestIdKey),
 		"topic_id":                     id,
+		"function":                     "getTopicHandler",
 	}
 
 	// get topic from mongoDB by id
-	topic, err := api.mongoDB.GetTopic(req.Context(), id)
+	topic, err := api.dataStore.Backend.GetTopic(req.Context(), id)
 	if err != nil {
 		handleError(ctx, w, err, logdata)
 		return
 	}
 
-	if err := WriteJSONBody(ctx, topic, w, logdata); err != nil {
+	// Ensure the sub document has the main document ID
+	topic.Current.ID = topic.ID
+
+	// User is not authenticated and hence has only access to current sub document
+	if err := WriteJSONBody(ctx, topic.Current, w, logdata); err != nil {
+		return
+	}
+	log.Event(ctx, "request successful", log.INFO, logdata) // NOTE: name of function is in logdata
+}
+
+// getTopicPrivateHandler is a handler that gets a topic by its id from MongoDB
+func (api *API) getTopicPrivateHandler(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	vars := mux.Vars(req)
+	id := vars["id"]
+	hColID := ctx.Value(handlers.CollectionID.Context())
+	logdata := log.Data{
+		handlers.CollectionID.Header(): hColID,
+		"request_id":                   ctx.Value(dprequest.RequestIdKey),
+		"topic_id":                     id,
+		"function":                     "getTopicPrivateHandler",
+	}
+
+	//logData := log.Data{}
```

Comment:
I think these comments can now be removed, as you said:
`getTopicPrivateHandler` is wrapped in api.go by `isAuthorised`, which is wrapped by `isAuthenticated`, isAuthenticated calls `CheckIdentity` in dp-net, which validates that the `CallerIdentity` is present in the request context. Then `isAuthorised` validates the permissions.

Reply:
Thanks.
now removed.